openc3/python/test/packets/test_packet.py

```diff
@@ -2042,3 +2043,111 @@ def test_subpacket_attribute_can_be_set(self):
     def test_subpacketizer_attribute_defaults_to_none(self):
         p = Packet("tgt", "pkt")
         self.assertIsNone(p.subpacketizer)
+
+    def test_subpacketize_item(self):
+        json_struct = Packet()
+        json_struct.accessor = JsonAccessor(json_struct)
+        json_struct.virtual = True
+        json_template = bytearray('{"id_item":1,"item1":101,"more":{"item2":12,"item3":3.14,"item4":"Example","item5":[]}}', "ascii")
+        json_struct.template = json_template
+        item = json_struct.append_item("ITEM0", 32, 'INT')
+        item.default = 1
+        item.key = "$.id_item"
+        item = json_struct.append_item("ITEM1", 16, 'UINT')
+        item.default = 101
+        item.key = "$.item1"
+        item = json_struct.append_item("ITEM2", 16, 'UINT')
+        item.default = 12
+        item.key = "$.more.item2"
+        item = json_struct.append_item("ITEM3", 64, 'FLOAT')
+        item.default = 3.14
+        item.key = "$.more.item3"
+        item = json_struct.append_item("ITEM4", 128, 'STRING')
+        item.default = "Example"
+        item.key = "$.more.item4"
+        item = json_struct.append_item("ITEM5", 8, 'UINT')
+        item.array_size = 0
+        item.default = []
+        item.key = "$.more.item5"
+
+        cbor_struct = Packet()
+        cbor_struct.accessor = CborAccessor(cbor_struct)
+        cbor_struct.virtual = True
+        cbor_template = dumps({"id_item":2, "item1":101, "more": { "item2":12, "item3":3.14, "item4":"Example", "item5":[] } })
+        cbor_struct.template = cbor_template
+        item = cbor_struct.append_item("ITEM0", 32, 'INT')
+        item.default = 2
+        item.key = "$.id_item"
+        item = cbor_struct.append_item("ITEM1", 16, 'UINT')
+        item.default = 101
+        item.key = "$.item1"
+        item = cbor_struct.append_item("ITEM2", 16, 'UINT')
+        item.default = 12
+        item.key = "$.more.item2"
+        item = cbor_struct.append_item("ITEM3", 64, 'FLOAT')
+        item.default = 3.14
+        item.key = "$.more.item3"
+        item = cbor_struct.append_item("ITEM4", 128, 'STRING')
+        item.default = "Example"
+        item.key = "$.more.item4"
+        item = cbor_struct.append_item("ITEM5", 8, 'UINT')
+        item.array_size = 0
+        item.default = []
+        item.key = "$.more.item5"
+
+        packet = Packet()
+        item = packet.append_item("JSON_LENGTH", 32, 'UINT')
+        item.default = 33
+        item = packet.append_item("JSON", 264, 'BLOCK')
+        item.default = ""
+        item.variable_bit_size = {'length_item_name': "JSON_LENGTH", 'length_bits_per_count': 8, 'length_value_bit_offset': 0}
```

Comment:
I tried to remove the `JSON_LENGTH` (lines 2099, 2100, 2103) and when `packet.restore_defaults` is executed I get `ValueError: value of 87 bytes does not fit into 33 bytes for data_type BLOCK`. But the `append_item` is clearly saying it is 264 bits (33 bytes) so not sure how the length field plays into this.